### PR TITLE
Handle initializers in printer

### PR DESCRIPTION
**Pull request by gramalingam**
* Update parser documentation to show syntax for initializers and value-infos
* Update printer to print initializers and value-infos
(See https://github.com/onnx/onnx/issues/5529 )

### CODE REVIEW

File: onnx/defs/printer.cc [hunk 1]
```diff
@@ -186,14 +186,17 @@ void ProtoPrinter::print(const TypeProto& type) {
...
   output_ << PrimitiveTypeNameMap::ToString(tensor.data_type());
   if (tensor.dims_size() > 0)
     printSet("[", ",", "]", tensor.dims());
 
   if (!tensor.name().empty()) {
     output_ << " " << tensor.name();
   }
+  if (is_initializer) {
+    output_ << " = ";
+  }
```
Comment (jcwchen) [inline onnx/defs/printer.cc:199]: So if the given TensorProto is an initializer, the print function will just add a equal? Do you have an example what will be the printer like for initializer and non-initializer here?

**Review by jcwchen — APPROVED** [review]
Not sure whether it is worthy to add tests for printer though.

File: docs/Syntax.md [hunk 2]
```diff
@@ -57,6 +57,11 @@ The grammar below describes the syntax:
    value-info ::= type id
    value-infos ::= value-info (',' value-info)*
    value-info-list ::= '(' value-infos? ')'
+   value-info-or-initializer ::= type id [ = '{' prim-constants '}']
```
Comment (justinchuby) [inline docs/Syntax.md:60]: How will the raw data be represented?

File: onnx/defs/printer.cc (hunk 1 above)
Comment (gramalingam) [reply onnx/defs/printer.cc:199]: Yes. We are reusing the same code for two contexts. The syntax for initializer requires the " = ", but a TensorProto, used as an attribute value does not. The printer follows the exact syntax supported by the parser.

Example of initializer is here in [this parser test](https://github.com/onnx/onnx/blob/e1d921373fe58a22d62fa861daff1c22e279fb8a/onnx/test/cpp/parser_test.cc#L305).

Example of a non-initializer is here in [this parser test](https://github.com/onnx/onnx/blob/e1d921373fe58a22d62fa861daff1c22e279fb8a/onnx/test/cpp/parser_test.cc#L174)

And the parser test invokes the printer, and parses it back, to ensure round-tripping works [here](https://github.com/onnx/onnx/blob/e1d921373fe58a22d62fa861daff1c22e279fb8a/onnx/test/cpp/parser_test.cc#L23)

**Pull request merged**